### ISSUE TITLE
Missing the GlobalDCE pass after the RepoPruning pass when using opt with no optimisation level.

### ISSUE THREAD

**Issue by MaggieYingYi** [SNSystems/llvm-project-prepo#103]
Compiling test.ll:
```
define void @f() {
  ret void
}
```
Using:
```
opt -mtriple="x86_64-pc-linux-gnu-repo" -debug-pass=Structure test.ll -o /dev/null
```
Command output:
```
Pass Arguments:  -targetlibinfo -tti -targetpassconfig -prepo -prepo-pruning -verify -write-bitcode
Target Library Information
Target Transform Information
Target Pass Configuration
  ModulePass Manager
    RepoMetadataGenerationPass
    RepoPruningPass
    FunctionPass Manager
      Module Verifier
    Bitcode Writer
```
We can see the GlobalDCEPass is missing after the RepoPruningPass.

**Comment by paulhuggett** [comment]
> We can see the GlobalDCEPass is missing after the RepoPruningPass.

I can certainly see that GlobalDCEPass is not present in the output you show. 

Where should it be in the pipeline?
Is this a new bug or a regression?

**Comment by MaggieYingYi** [comment]
> 
> Where should it be in the pipeline?
>
After fixing the issue#https://github.com/SNSystems/llvm-project-prepo/issues/60 by using the PR#https://github.com/SNSystems/llvm-project-prepo/pull/61, the pipeline of the passes should be `RepoMetadataGenerationPass`, `RepoPruningPass` and `Dead Global Elimination`.

>
> Is this a new bug or a regression?
>
When I worked on issue60, I added the `Dead Global Elimination` pass after the `Repo Rruning` pass for O0, O1, O2, O3 and Os optimisation level. However, I forgot to add the `Dead Global Elimination` pass when using `opt` command without giving any optimisation level option. This results in current issue.

>
> Other than missed optimizations, are there any other consequences of this problem?
>
If some dead global objects could be but not be removed, they still need to go through all unnecessary rest of passes. This will result in more compilation time.
